src/SFA.DAS.EmployerIncentives.Web/Services/BankDetails/IBankDetailsStatusService.cs

```diff
@@ -0,0 +1,10 @@
+﻿using System.Threading.Tasks;
+using SFA.DAS.EmployerIncentives.Web.Models;
+
+namespace SFA.DAS.EmployerIncentives.Web.Services.BankDetails
+{
+    public interface IBankDetailsStatusService
+    {
+        Task<LegalEntityModel> RecordBankDetailsComplete(string accountId, string accountLegalEntityId);
```

Comment:
Why would a method that updates return an unrelated object?  This doesn't follow the command and query separation pattern.  

Do we need a bank details status service?  This seems very granular, I would expect it to act on a BankDetailsStatus object.  There's already a bank details service and an application service so wouldn't it be better on one of those interfaces?  Isn't the status change against the application?  So would be better on that interface perhaps?

Reply:
The status is recorded against the legal entity, so will move into LegalEntitiesService